src/Input/ToggleSwitch/ToggleSwitch.test.tsx

```diff
@@ -88,6 +109,43 @@ describe('<ToggleSwitch /> click behavior', () => {
     expect(toggleSwitchContainer).toHaveAttribute('aria-checked', 'true');
   });
 
+  it('should change aria-checked correctly when it is controlled', () => {
+    let active = false;
+    const onChangeSpy = jest.fn(isActive => {
+      active = isActive;
+    });
+
+    console.log('active', active);
+
+    const { getByRole, rerender } = render(
+      <ToggleSwitch
+        active={active}
+        iconOptions={{
+          active: EyeIcon,
+          inactive: EyeSlashedIcon,
+        }}
+        onChange={onChangeSpy}
+      />
+    );
+    const toggleSwitchContainer = getByRole('switch');
+    expect(toggleSwitchContainer).toHaveAttribute('aria-checked', 'false');
+
+    userEvent.click(toggleSwitchContainer);
+    expect(onChangeSpy).toHaveBeenCalledTimes(1);
+
+    rerender(
```

Comment:
Oh why do we have to rerender it? I assume after onChangeSpy is called the aria-checked will be updated?

Reply:
Because ToggleSwitch is controlled, it should only be updated when `active` prop is changed.